state/state.go

```diff
@@ -407,8 +412,25 @@ func (s *State) isExtracted(p *manifest.Package) bool {
 }
 
 func (s *State) areBinariesLinked(p *manifest.Package) bool {
-	_, err := os.Stat(filepath.Join(s.binaryDir, p.Reference.String()))
-	return err == nil
+	if _, err := os.Stat(filepath.Join(s.binaryDir, p.Reference.String())); err != nil {
+		return false
+	}
+
+	bins, err := p.ResolveBinaries()
+	if err != nil {
+		return false
+	}
+
+	binsDir := filepath.Join(s.binaryDir, p.Reference.String())
+
+	for _, bin := range bins {
```

Comment:
This code is already in `linkBinaries()`, but also it shouldn't be in this function. I thought the plan was to just check if the symlinks existed then let the normal linking occur?

Reply:
oh shit thanks I misimplemented! 😅 I wrote this right after waking up.

Reply:
Fixed. Let me know if this implementation works for you. ❤️